substrate/tokens/pallet/src/lib.rs

```diff
@@ -37,7 +37,7 @@ pub mod pallet {
   pub struct Pallet<T>(PhantomData<T>);
 
   impl<T: Config> Pallet<T> {
-    fn burn_internal(
+    pub fn burn_internal(
```

Comment:
My general convention is when we have multiple paths over the same core functionality, to add the two paths *with a single `_internal` execution block*. The higher paths sometimes just fill in any pre-set variables, *yet sometimes do path specific execution*.

I'm not an advocate for making an `_internal` function `pub` unless we make it not an `_internal` function. We'd have to

1) Name it something else, which shows it's a complete function (not one with a prerequisites/assumptions on the caller)
2) Double check its current calls

As for the name, I'd suggest calling it burn but my guess is that'd conflict with the extrinsic-reachable Call? It may not due to the macro transformations which Call undergoes? Feel free to try renaming this to burn.

Reply:
doesn't compile with naming it `burn.  `duplicate definitions`  error so yea prob have to name it something else.